src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -758,5 +756,58 @@ public void It_does_not_write_to_undefined_assembly_metadata_attribute(string ta
                 AssemblyInfo.Get(assemblyPath).ContainsKey("AssemblyMetadataAttribute").Should().Be(false);
             }
         }
+
+        [Theory]
+        [InlineData("netcoreapp3.1", ".NET Core 3.1")]
+        [InlineData("netcoreapp2.1", ".NET Core 2.1")]
+        [InlineData("netstandard2.1", ".NET Standard 2.1")]
+        [InlineData("net5.0", ".NET 5.0")]
+        public void CheckTargetFrameworkDisplayName(string targetFrameworkVersion, string expectedFrameworkDisplayName)
+        {
+            TestProject libraryProject = new TestProject()
+            {
+                Name = "LibraryProject",
+                TargetFrameworks = targetFrameworkVersion
+            };
+            libraryProject.SourceFiles["Class.cs"] = @"
+public class LibraryClass{}
+";
+
+            TestProject testProject = new TestProject()
+            {
+                Name = "HelloWorld",
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+                IsExe = true
+            };
+
+            testProject.ReferencedProjects.Add(libraryProject);
+            testProject.SourceFiles["Program.cs"] = @"
+using System;
+using System.Reflection;
+using System.Runtime.Versioning;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var str = typeof(LibraryClass).Assembly.GetCustomAttribute<TargetFrameworkAttribute>().FrameworkDisplayName;
+        Console.WriteLine(str);
+    }
+}";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
```

Comment:
Pass in a value for the identifier parameter, you can use the `targetFrameworkVersion` parameter that will ensure the test directory hash is more unique and avoid the issues you're seeing in Helix I think